pkg/resource/replication_group/custom_set_output.go

```diff
@@ -297,3 +304,54 @@ func (rm *resourceManager) setLastRequestedLogDeliveryConfigurations(
 		annotations[AnnotationLastRequestedLDCs] = string(lastRequestedConfigs)
 	}
 }
+
+// setLastRequestedCacheNodeType copies desired.Spec.CacheNodeType into the annotation
+// of the object.
+func (rm *resourceManager) setLastRequestedCacheNodeType(
+	r *resource,
+	annotations map[string]string,
+) {
+	if r.ko.Spec.CacheNodeType != nil {
+		annotations[AnnotationLastRequestedCNT] = *r.ko.Spec.CacheNodeType
+	}
+}
+
+// setLastRequestedNodeGroupConfiguration copies desired.spec.NodeGroupConfiguration into the
+// annotation of the object
+func (rm *resourceManager) setLastRequestedNodeGroupConfiguration(
+	r *resource,
+	ko *svcapitypes.ReplicationGroup,
+) {
+	desiredAnnotations := r.ko.ObjectMeta.GetAnnotations()
+	annotations := make(map[string]string)
+	for k, v := range desiredAnnotations {
+		annotations[k] = v
+	}
```

Comment:
Is it necessary to copy the map every time we need to set one annotation? maybe we can just initiated ones and set the annotations?